backend/src/main/java/com/rota/database/orm/staff/Staff.java

```diff
@@ -44,6 +45,7 @@
 
   private String jobTitle;
 
+  @Column(unique = true)
```

Comment:
This should be enough to raise an exception if user wants to save a duplicate e-mail.
I'm wondering if it's enough to convert an SQL exception to DuplicateEmailException.
That way would save us one database lookup every time we add a new user.
Not a big deal either way, because I don't expect anyone to be spamming 'add user'.

Reply:
Initially I did this - I caught the ConstraintViolationException and threw a DuplicateEmailException but wasn't sure if that was a good idea, as other constraints may be broken and then we have a bad error message?

Reply:
Good point. Currently it is our only constraint, but we shouldn't rely on that.

Reply:
![image](https://user-images.githubusercontent.com/8884999/82314839-4e9f7680-99c2-11ea-981b-681fd31b37af.png)

This the current response when updating a staff member with an unavailable email currently. I could make the error handler check the message and decide if it is email related and send the correct exception? And if it can't decide what it is, throw another, less specific exception that we can dig into?

Reply:
Someone more experienced should comment on this, you COULD search the string for `PUBLIC.STAFF(EMAIL)` for example, but this feels hacky and I'm worried it would make the code harder to read, so I'd be inclined to keep it as it is.

Reply:
Yeah I thought about this but wasn't sure if it was the best approach. We should leave it as it is and then when testers have a look we can decide what's best.